### PR TITLE
Migrate Sources of TransistorPublishPlugin to this package

### PR DESCRIPTION
**Pull request by devahmedshendy**

### CODE REVIEW

File: Sources/TransistorPublishPlugin/TransistorPublishPlugin.swift [hunk 1]
```diff
@@ -1,2 +1,52 @@
-// The Swift Programming Language
-// https://docs.swift.org/swift-book
+// swiftlint:disable:this file_name
```
Comment (leogdion) [inline Sources/TransistorPublishPlugin/TransistorPublishPlugin.swift]: rename to Plugin

File: Sources/TransistorPublishPlugin/TransistorEmbedGenerator.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  /// Generates an EmbeddedTransistor from the URL.
+  public func generate() -> EmbeddedTransistor {
+    EmbeddedTransistor(html: html(from: transistorURL.lastPathComponent))
+  }
+
+  /// Creates an HTML embed code for a podcast id.
+  private func html(from id: String) -> String {
+    // swiftlint:disable line_length
```
Comment (leogdion) [inline Sources/TransistorPublishPlugin/TransistorEmbedGenerator.swift]: Can this be broken up and still work?

File: Sources/TransistorPublishPlugin/EmbeddedTransistor.swift [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import Foundation
+
+/// A type for holding the HTML embed code for a Transistor podcast.
+public struct EmbeddedTransistor: Hashable, Codable {
```
Comment (leogdion) [inline Sources/TransistorPublishPlugin/EmbeddedTransistor.swift:4]: Does this need to be public?

File: Sources/TransistorPublishPlugin/EmbeddedTransistor.swift [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+  public let html: String
```
Comment (leogdion) [inline Sources/TransistorPublishPlugin/EmbeddedTransistor.swift:6]: Add comment if this remains public

File: Sources/TransistorPublishPlugin/TransistorPublishPlugin.swift [hunk 5]
```diff
@@ -1,2 +1,52 @@
...
+    renderer: TransistorRenderer = .default
+  ) -> Self {
+    Plugin(name: "Transistor") { context in
+      context.markdownParser.addModifier(.transistorBlockQuote(using: renderer))
+    }
+  }
+}
+
+/// An modifier for Ink that helps rendering a Transistor blockquotes into HTML string.
+public extension Modifier {
```
Comment (leogdion) [inline Sources/TransistorPublishPlugin/TransistorPublishPlugin.swift]: put in separate file `Modifier`

File: Sources/TransistorPublishPlugin/TransistorRenderer.swift [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+/// A default implementation for TransistorRenderer.
+/// It just provides the html string hold by the transistor.
+public struct DefaultTransistorRenderer: TransistorRenderer {
```
Comment (leogdion) [inline Sources/TransistorPublishPlugin/TransistorRenderer.swift]: put in separate file with correct name

File: Sources/TransistorPublishPlugin/TransistorEmbedGenerator.swift (hunk 2 above)
Comment (devahmedshendy) [reply Sources/TransistorPublishPlugin/TransistorEmbedGenerator.swift]: You mean the embed code string, to avoid disabling lint error?

**Pull request merged**